### PR TITLE
Adapt serverless functions to be run in non-privileged mode

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- add `serverless-function` psp which enables istioInit sidecar with root user
- separate `serverless` serviceaccount into two, one for kaniko only and one for functions

**Related issue(s)**

#9785

### CODE REVIEW

**Review by tgorgol — COMMENTED** [review]
First part of review

File: resources/serverless/templates/service-accounts.yaml [hunk 1]
```diff
@@ -9,12 +9,18 @@ metadata:
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ template "fullname" . }}
+  name: {{ template "fullname" . }}-function
   namespace: {{ .Release.Namespace }}
   labels:
     serverless.kyma-project.io/config: service-account
 automountServiceAccountToken: false
-secrets:
```
[the remark this answers is not in the document]
Comment (aerfio) [reply resources/serverless/templates/service-accounts.yaml:17]: Yup, looks like we dont, great reasearch 👍🏻

**Review by tgorgol — APPROVED** [review]

**Pull request merged**